airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -301,3 +304,138 @@ async def _build_connector_acceptance_test(self, connector_under_test_container:
             )
 
         return cat_container.with_unix_socket("/var/run/docker.sock", self.context.dagger_client.host().unix_socket("/var/run/docker.sock"))
+
+
+class RegressionTests(Step):
+    """A step to run regression tests for a connector."""
+
+    context: ConnectorContext
+    title = "Regression tests"
+    skipped_exit_code = 5
+    accept_extra_params = True
+    regression_tests_artifacts_dir = Path("/tmp/regression_tests_artifacts")
+    working_directory = "/app"
+
+    @property
+    def default_params(self) -> STEP_PARAMS:
+        """Default pytest options.
+
+        Returns:
+            dict: The default pytest options.
+        """
+        return super().default_params | {
+            "-ra": [],  # Show extra test summary info in the report for all but the passed tests
+            "--disable-warnings": [],  # Disable warnings in the pytest report
+            "--durations": ["3"],  # Show the 3 slowest tests in the report
+        }
+
+    def regression_tests_command(self, start_timestamp: int) -> List[str]:
+        return [
+            "poetry",
+            "run",
+            "pytest",
+            "src/live_tests/regression_tests",
+            "--connector-image",
+            self.connector_image,
+            "--connection-id",
+            self.connection_id,
+            "--control-version",
+            self.control_version,
+            "--target-version",
+            self.target_version,
+            "--pr-url",
+            self.pr_url,
+            "--start-timestamp",
+            str(start_timestamp),
+            "--should-read-with-state",
+            str(self.should_read_with_state),
+        ]
+
+    def __init__(self, context: ConnectorContext) -> None:
+        """Create a step to run regression tests for a connector.
+
+        Args:
+            context (ConnectorContext): The current test context, providing a connector object, a dagger client and a repository directory.
+        """
+        super().__init__(context)
+        self.connector_image = context.docker_image.split(":")[0]
+        print(self.context.run_step_options.step_params)
+        options = self.context.run_step_options.step_params.get(CONNECTOR_TEST_STEP_ID.CONNECTOR_REGRESSION_TESTS, {})
+
+        self.connection_id = self._get_item_or_default(options, "connection-id", None)
+        self.pr_url = self._get_item_or_default(options, "pr-url", None)
+
+        if not self.connection_id and self.pr_url:
+            raise ValueError("`connection-id` and `pr-url` are required to run regression tests.")
+
+        self.control_version = self._get_item_or_default(options, "control-version", "latest")
+        self.target_version = self._get_item_or_default(options, "target-version", "dev")
+        self.should_read_with_state = self._get_item_or_default(options, "should-read-with-state", True)
+
+    @staticmethod
+    def _get_item_or_default(options: Dict[str, Any], key: str, default: Any):
```

Comment:
I don't think so but it can go to the helpers / utils. I'd be more inclined to use `dpath` instead of writing this kind of function.